### ISSUE TITLE
Tunnel access is not working in docker container M1

### ISSUE THREAD

**Issue by tealcake2** [docker/for-mac#5374]
Hi,

on apple M1 Tech preview,

I trying to connect to mysql server with ssh tunnel, but it is not working. 

When I trying connect to tunnel from container with `host.docker.internal`, I got connection refused.

on another pc without M1 version it is working fine

**Comment by stephen-turner** [comment]
mysql doesn't work properly on M1 because there isn't an arm64 build of it. In the documentation (https://docs.docker.com/docker-for-mac/apple-m1/) we suggest using mariadb instead.

**Comment by tealcake2** [comment]
??? this is totaly diffrent issue ... i do not have problem with mysql container, I just can not connect to tunneled connection ip 

So when I create ssh tunnel connection in my macos to production mysql, for example to port 23306, then I do not have access to it from docker container, for example from php-fpm container I can not connect to mysql with host host.docker.internal:23306

**Comment by goyan** [comment]
agreed,  issues is connection refused when connecting to host process by port.
on my side, I can't access to host process using ports, 8001, 5432 

docker.host.internal resolves to an ip,  pings correctly from docker, but connection is refused on processes by port.

**Comment by beijaflor** [comment]
I have same problem here. I can't connect hosts 9200 port which connect to GCP compute engine via ssh tunneling.
While trying to solve this problem, I found that It's also happen when to connect ssh tunneling port of other container in same network bridge. It may not only problem with host to container connection, but entire network.

For work around, I make connection inside the container and it works. 
But it's really ugly and not-handy solution. Looking forward to solve this issue...

**Comment by leandregagnonlewis** [comment]
Same problem! When i look in /etc/hosts there is no added host for host.docker.internal so it is impossible to resolve.
`root@bc81cc5b3bce:/home# cat /etc/hosts
127.0.0.1	localhost
::1	localhost ip6-localhost ip6-loopback
fe00::0	ip6-localnet
ff00::0	ip6-mcastprefix
ff02::1	ip6-allnodes
ff02::2	ip6-allrouters
172.17.0.2	bc81cc5b3bce`

**Comment by ryanoillataguerre** [comment]
Hello! I'm running into the same issue on an M1 when using tunnels specifically. I'm tunneling to a postgres DB on `localhost:5432`, which is reachable using `psql` from my local, but when I pass the same connection string as used in the terminal into the docker app as an ENV var, it refuses connection.

FWIW, `host.docker.internal:5432` is being translated to `192.168.64.1:5432` internally.

Our entire `docker-compose` dev workflow revolves around this tunneling into our DB, so would really like to know if there's a valid workaround. Haven't been able to find anything on support forums that's worked.

**Comment by nitinv80** [comment]
Hi, I am also running the same issue. IS there any solution or patch found?

**Comment by aaronfagan** [comment]
Also seeing this issue now Reverse Proxying (Container) to Plex (Host).

**Comment by jmarbutt** [comment]
I am having this issue also, any updates?
